src/Geocoding/CachedGeocodingService.php

```diff
@@ -24,9 +24,11 @@ public function __construct(GeocodingService $geocodingService, Cache $cache)
         $this->cache = $cache;
     }
 
-    public function getCoordinates(string $address): ?Coordinates
+    public function getCoordinates(string $address, string $locationName=''): ?Coordinates
     {
-        $encodedCacheData = $this->cache->fetch($address);
+        $key = empty($locationName) ? $address : trim($address . $address);
```

Comment:
This feels like a copy/paste error `trim($address . $address);`

Reply:
Also can the `getKey` not be used from the injected `geocodingService`?

Reply:
I will also add a unit test for testing the key

Reply:
I renamed the function, added a unit test, and also made it reuse the method searchTerm() now required for the GeocodingService interface. I did this because the logic how to make the key was duplicate code.